components/editor/editor/CodeBlock.tsx

```diff
@@ -0,0 +1,52 @@
+import { NodeViewContent, NodeViewWrapper } from "@tiptap/react";
+import { ChangeEvent, FunctionComponent } from "react";
+
+// Change this for code styling
+import "highlight.js/styles/monokai-sublime.css";
+
+interface CodeBlockProps {
+  node: {
+    attrs: {
+      language: string;
+    };
+  };
+  updateAttributes: (attributes: { language: string }) => void;
+  extension: any; // Extension Files - usually unknown
+}
+const CodeBlock: FunctionComponent<CodeBlockProps> = ({
+  node: {
+    attrs: { language: defaultLanguage },
+  },
+  updateAttributes,
+  extension,
+}) => {
+  return (
+    <NodeViewWrapper className="code-block">
+      <select
+        onClick={(e) => e.stopPropagation()}
+        onMouseDown={(e) => e.stopPropagation()}
+        contentEditable={false}
+        defaultValue={defaultLanguage}
+        onChange={(event: ChangeEvent<HTMLSelectElement>) => {
+          updateAttributes({ language: event.target.value });
+        }}
+      >
+        <option value="null">auto</option>
+        <option disabled>—</option>
+        {extension.options.lowlight
+          .listLanguages()
+          .map((lang: string, index: number) => (
+            <option key={index} value={lang}>
+              {lang}
+            </option>
+          ))}
+      </select>
+
+      <pre className="bg-black border ">
```

Comment:
@NiallJoeMaher are there styles for code blocks at the moment? may be easier to add here.

Reply:
The styles we currently have are great; I wonder if we can mimic or what people think 🤔

Reply:
We should be able to - We can add this and figure it out - if it doesnt work we can revert the changes and save syntax highlighting for v2 of the editor